.gitignore

```diff
@@ -1,3 +1,6 @@
 *.hi
 *.o
 stack.yaml
+dist
+.*
+!/.gitignore
```

Comment:
It does not look right to me to ignore all dot-files. For instance, this setup ignores `.travis.yml` as well.